### PR TITLE
remove retention policy for staging buckets

### PR DESCRIPTION
**Pull request by ameukam**
Follow-up of:
  - https://github.com/kubernetes/kubernetes/pull/115634

The current retention policy prevent creation or update of new objects until the existing one are deleted based on the retention period.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ameukam** [comment]
Any new bucket should be created without retention policy.

**Comment by aojea** [comment]
👀

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -333,7 +333,7 @@ function upload-tars() {
     # Ensure the buckets are created
     if ! gsutil ls "${staging_bucket}" >/dev/null; then
       echo "Creating ${staging_bucket}"
-      gsutil mb -l "${region}" -p "${PROJECT}" --retention "${retention_period}" "${staging_bucket}"
+      gsutil mb -l "${region}" -p "${PROJECT}" "${staging_bucket}"
```
Comment (BenTheElder) [inline cluster/gce/util.sh:336]: probably we should add `gsutil retention clear "${staging_bucket}"` below this if block?

Comment (ameukam) [reply cluster/gce/util.sh:336]: Preferable to do this manually to quickly unblock ?

Comment (BenTheElder) [reply cluster/gce/util.sh:336]: Only if it's actually easier to enumerate the buckets than adding this one line?

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/pull/116197#issuecomment-1452238981

when we circle back to this we will want to set a lifecycle policy (TTL rule) instead of retention policy (forced to keep) 😅

**Comment by BenTheElder** [comment]
Real verify failure:
```console
In ./cluster/gce/util.sh line 329:
    retention_period="1d" # https://cloud.google.com/storage/docs/bucket-lock#retention-periods
    ^--------------^ SC2034 (warning): retention_period appears unused. Verify use (or export if used externally).
```
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116222/pull-kubernetes-verify/1631339053372674048

Recommend adding a `gsutil retention clear` while fixing this.

File: cluster/gce/util.sh (hunk 1 above)
Comment (dims) [reply cluster/gce/util.sh:335]: +1 add the line Ben mentions.

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**